### ISSUE TITLE
canonical URLs

### ISSUE THREAD

**Issue by salahjaafar** [templaza/astroid-framework#438]
How to add canonical URLs has become a mandatory part of web SEO on Google

**Comment by salahjaafar** [comment]
My experience shows that Google won't index pages without a canonical URL.

I'm using the Helix framework on some sites, and it handles canonical URLs automatically.

I want to develop my future websites with the Astroid framework, but I need to ensure the SEO side.

**Comment by sonvnn** [comment]
Thank you for your suggestion. I will add this feature in v3.0.4. I will inform you as soon as this feature is available then you can get update first before 3.0.4 is released

**Comment by sonvnn** [comment]
@salahjaafar @Chacapamac After research about Canonical url. I think it will be good if it was generated by 3rd party extensions. You can find many here https://extensions.joomla.org/tags/seo-metadata/ They will do better than us. I research Helix Ultimate then it does not support Canonical, too.

**Comment by salahjaafar** [comment]
The automatic generation of canonical URLs is the most efficient, and effective extensions are paid. Furthermore, adding this functionality to our framework increases usage and popularity.

**Comment by Chacapamac** [comment]
I agree that even if that Google Guy, (on my previous post), to my surprise, says that it is “Not a Factor for Negative Ranking” All SEO experts prefer Canonicals URLS as they are ALWAYS treated by Google as unique page/content and treated as such in its analytical records. If Google encounters quasi-identical non-canonical page/content it will choose one of them as the “REAL” content and will simply bypass the other similar page/content in its analytical process. Canonical should be preferred for similar pages/content that NEED to be recorded by Google Analytics as separated entity.

**Comment by sonvnn** [comment]
I will research 3rd party extension to understand how they work. After that I will consider to add this to Astroid in next versions.

Thanks for feedback.

**Comment by salahjaafar** [comment]
Thanks for your efforts, I believe the developers of Joomla are primarily designed for this modification.
At the same time, the Astroid framework team is included in this evolution, because user community is important
